backend/de.metas.business/src/main/java/de/metas/invoice/service/impl/AbstractInvoiceDAO.java

```diff
@@ -589,4 +590,22 @@ public boolean hasCompletedInvoicesReferencing(@NonNull final InvoiceId invoiceI
 				.create()
 				.listIds(lineId -> InvoiceLineId.ofRepoId(id, lineId));
 	}
+
+	private boolean matchesDocType(@NonNull final I_C_Invoice serviceFeeInvoiceCandidate, @Nullable final DocBaseAndSubType targetDocType)
+	{
+		if (targetDocType == null)
+		{
+			return true;
+		}
+
+		final DocTypeId docTypeId = CoalesceUtil.coalesceNotNull(
+				DocTypeId.ofRepoIdOrNull(serviceFeeInvoiceCandidate.getC_DocType_ID()),
+				DocTypeId.ofRepoId(serviceFeeInvoiceCandidate.getC_DocTypeTarget_ID()));
+
+		final I_C_DocType docTypeRecord = docTypeDAO.getById(docTypeId);
+
+		return docTypeRecord.getDocBaseType().equals(targetDocType.getDocBaseType())
```

Comment:
I would prefer to extract the DocBaseAndSubType from I_C_DocType and compare it with `targetDocType`.
In this way we would cover the case when `docSubType` is null on both sides.